enterprise_access/apps/api/v1/views/browse_and_request.py

```diff
@@ -817,9 +837,49 @@ def create(self, request, *args, **kwargs):
             return Response({"detail": exc.message}, status=exc.http_status_code)
 
         policy = self.request.validated_policy
+        course_id = self.request.data.get("course_id")
+        course_price = self.request.data.get("course_price")
+
+        # Check if learner is re-requesting the same course under the same policy.
+        existing_request = LearnerCreditRequest.objects.filter(
+            user__lms_user_id=request.user.lms_user_id,
+            learner_credit_request_config=policy.learner_credit_request_config,
+            course_id=course_id,
+        ).first()
+
+        # If an existing request is found in CANCELLED, EXPIRED or REVERSED state we
+        # reuse it instead of creating a new one.
+        if existing_request and existing_request.state in REUSABLE_REQUEST_STATES:
+            try:
+                self._reuse_existing_request(existing_request, course_price)
+                # Trigger admin email notification with the latest request
+                send_learner_credit_bnr_admins_email_with_new_requests_task.delay(
```

Comment:
[Thoughts] I think we should trigger an email after logging actions. [Reason] If an error occur while logging errors then the sent email will be a complete false positive.